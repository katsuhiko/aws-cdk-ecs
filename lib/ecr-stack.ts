import { Construct, Stack, StackProps } from '@aws-cdk/core';
import { Repository } from '@aws-cdk/aws-ecr';

export class EcrStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    new Repository(this, 'WebRepo', {
      repositoryName: 'demo-cdk-web',
      imageScanOnPush: true,
      lifecycleRules: [
        { description: 'Delete old images', maxImageCount: 5 },
      ],
    });

    new Repository(this, 'AppRepo', {
      repositoryName: 'demo-cdk-app',
      imageScanOnPush: true,
      lifecycleRules: [
        { description: 'Delete old images', maxImageCount: 5 },
      ],
    });
  }
}
