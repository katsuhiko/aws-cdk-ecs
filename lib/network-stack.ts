import { Construct, Stack, StackProps } from '@aws-cdk/core';
import { SubnetType, Vpc } from '@aws-cdk/aws-ec2';

export class NetworkStack extends Stack {

  public readonly vpc: Vpc

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    this.vpc = new Vpc(this, 'VPC', {
      maxAzs: 2,
      subnetConfiguration: [
        { name: 'Frontend', subnetType: SubnetType.PUBLIC, cidrMask: 24 },
        { name: 'Application', subnetType: SubnetType.PUBLIC, cidrMask: 24 },
        { name: 'Datastore', subnetType: SubnetType.ISOLATED, cidrMask: 24 },
      ],
    });
  }
}
