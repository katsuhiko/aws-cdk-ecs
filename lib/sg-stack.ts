import { Construct, Stack, StackProps, Tags } from '@aws-cdk/core';
import { IVpc, Peer, Port, SecurityGroup } from '@aws-cdk/aws-ec2';

interface SgStackProps extends StackProps {
  vpc: IVpc
}

export class SgStack extends Stack {

  public readonly bastionSg: SecurityGroup;
  public readonly albSg: SecurityGroup;
  public readonly webSg: SecurityGroup;
  public readonly rdsSg: SecurityGroup;

  constructor(scope: Construct, id: string, props: SgStackProps) {
    super(scope, id, props);

    this.bastionSg = new SecurityGroup(this, 'BastionSg', {
      vpc: props.vpc,
    });
    Tags.of(this.bastionSg).add('Name', 'SgStack/BastionSg');

    this.albSg = new SecurityGroup(this, 'AlbSg', {
      vpc: props.vpc,
    });
    Tags.of(this.albSg).add('Name', 'SgStack/AlbSg');
    this.albSg.addIngressRule(Peer.anyIpv4(), Port.tcp(80));
    this.albSg.addIngressRule(Peer.anyIpv4(), Port.tcp(443));

    this.webSg = new SecurityGroup(this, 'WebSg', {
      vpc: props.vpc,
    });
    Tags.of(this.webSg).add('Name', 'SgStack/WebSg');
    this.webSg.addIngressRule(this.albSg, Port.tcp(80));

    this.rdsSg = new SecurityGroup(this, 'RdsSg', {
      vpc: props.vpc,
    });
    Tags.of(this.rdsSg).add('Name', 'SgStack/RdsSg');
    this.rdsSg.addIngressRule(this.bastionSg, Port.tcp(3306));
    this.rdsSg.addIngressRule(this.webSg, Port.tcp(3306));
  }
}
