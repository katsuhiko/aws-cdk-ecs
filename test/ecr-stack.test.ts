import { App } from '@aws-cdk/core';
import { SynthUtils } from '@aws-cdk/assert';
import { EcrStack } from '../lib/ecr-stack';

test('ECR Stack', () => {
  const app = new App();
  const stack = new EcrStack(app, 'TestStack');

  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
