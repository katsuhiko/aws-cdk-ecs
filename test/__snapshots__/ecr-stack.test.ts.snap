// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ECR Stack 1`] = `
Object {
  "Resources": Object {
    "AppRepoB478A890": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ImageScanningConfiguration": Object {
          "ScanOnPush": true,
        },
        "LifecyclePolicy": Object {
          "LifecyclePolicyText": "{\\"rules\\":[{\\"rulePriority\\":1,\\"description\\":\\"Delete old images\\",\\"selection\\":{\\"tagStatus\\":\\"any\\",\\"countType\\":\\"imageCountMoreThan\\",\\"countNumber\\":5},\\"action\\":{\\"type\\":\\"expire\\"}}]}",
        },
        "RepositoryName": "demo-cdk-app",
      },
      "Type": "AWS::ECR::Repository",
      "UpdateReplacePolicy": "Retain",
    },
    "WebRepo9D56C0A3": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ImageScanningConfiguration": Object {
          "ScanOnPush": true,
        },
        "LifecyclePolicy": Object {
          "LifecyclePolicyText": "{\\"rules\\":[{\\"rulePriority\\":1,\\"description\\":\\"Delete old images\\",\\"selection\\":{\\"tagStatus\\":\\"any\\",\\"countType\\":\\"imageCountMoreThan\\",\\"countNumber\\":5},\\"action\\":{\\"type\\":\\"expire\\"}}]}",
        },
        "RepositoryName": "demo-cdk-web",
      },
      "Type": "AWS::ECR::Repository",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
